### PR TITLE
Use branches to make releases

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: dagger/release.go [hunk 1]
```diff
@@ -68,8 +70,11 @@ func (r *Replicated) Release(
 		WithWorkdir("/go/src/github.com/replicatedhq/replicated").
 		WithExec([]string{"git", "config", "user.email", "release@replicated.com"}).
 		WithExec([]string{"git", "config", "user.name", "Replicated Release Pipeline"}).
+		WithExec([]string{"git", "remote", "add", "tag", fmt.Sprintf("https://%s@github.com/replicatedhq/replicated.git", githubTokenPlaintext)}).
+		WithExec([]string{"git", "checkout", "-b", releaseBranchName}).
 		WithExec([]string{"git", "add", "pkg/version/build.go"}).
-		WithExec([]string{"git", "commit", "-m", fmt.Sprintf("Set version to %d.%d.%d", major, minor, patch)})
+		WithExec([]string{"git", "commit", "-m", fmt.Sprintf("Set version to %d.%d.%d", major, minor, patch)}).
+		WithExec([]string{"git", "push", "tag", releaseBranchName}).
```
[the remark this answers is not in the document]
Comment (divolgin) [reply dagger/release.go:77]: why not?

[the remark this answers is not in the document]
Comment (marccampbell) [reply dagger/release.go:81]: this is pushing the branch, right?

File: dagger/release.go [hunk 2]
```diff
@@ -68,23 +74,21 @@ func (r *Replicated) Release(
...
-	if err != nil {
-		return err
-	}
 	tagContainer := dag.Container().
 		From("alpine/git:latest").
 		WithMountedDirectory("/go/src/github.com/replicatedhq/replicated", updatedSource).
 		WithWorkdir("/go/src/github.com/replicatedhq/replicated").
-		WithExec([]string{"git", "remote", "add", "tag", fmt.Sprintf("https://%s@github.com/replicatedhq/replicated.git", githubTokenPlaintext)}).
 		With(CacheBustingExec([]string{"git", "tag", fmt.Sprintf("v%d.%d.%d", major, minor, patch)})).
 		With(CacheBustingExec([]string{"git", "push", "tag", fmt.Sprintf("v%d.%d.%d", major, minor, patch)}))
```
Comment (marccampbell) [inline dagger/release.go:93]: should you remove this?

File: dagger/release.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (divolgin) [reply dagger/release.go:81]: "tag" is the remote and `releaseBranchName` is the branch name

`git push tag release-0.83.11`

File: dagger/release.go (hunk 2 above)
Comment (divolgin) [reply dagger/release.go:93]: I think this is needed to push the tag

**Review by marccampbell — APPROVED** [review]

**Pull request merged**